Hippo.Web/ClientApp/src/components/Product/Products.tsx

```diff
@@ -0,0 +1,462 @@
+                  units: e.target.value,
+                };
+                setEditProductModel(model);
+                setReturn(model);
+              }}
+            />
+          </div>
+          <div className="form-group">
+            <label htmlFor="fieldInstallmentType">Installment Type</label>
+            <select
+              className="form-control"
+              id="fieldInstallmentType"
+              required
+              value={editProductModel.installmentType}
+              onChange={(e) => {
+                const model: ProductModel = {
+                  ...editProductModel,
+                  installmentType: e.target.value,
+                };
+                if (model.installmentType === "OneTime") {
+                  model.installments = 1;
+                }
+                if (
+                  model.installmentType === "Monthly" &&
+                  (model.installments === 1 || model.installments === 5)
+                ) {
+                  model.installments = 60;
+                }
+                if (
+                  model.installmentType === "Yearly" &&
+                  (model.installments === 1 || model.installments === 60)
+                ) {
+                  model.installments = 5;
+                }
+                setEditProductModel(model);
+                setReturn(model);
+              }}
+            >
+              <option value="OneTime">One Time</option>
+              <option value="Monthly">Monthly</option>
+              <option value="Yearly">Yearly</option>
+            </select>
+          </div>
+          {editProductModel.installmentType === "OneTime" ? null : (
+            <div className="form-group">
+              <label htmlFor="fieldInstallments">Installments</label>
+              <input
+                className="form-control"
+                id="fieldInstallments"
+                required
+                value={editProductModel.installments}
+                onChange={(e) => {
+                  const model: ProductModel = {
+                    ...editProductModel,
+                    installments: parseInt(e.target.value),
+                  };
+                  setEditProductModel(model);
+                  setReturn(model);
+                }}
+              />
+            </div>
+          )}
+          <div className="form-group">
+            <label htmlFor="lifeCycle">Life Cycle in Months</label>
+            <input
+              className="form-control"
+              id="lifeCycle"
+              required
+              value={editProductModel.lifeCycle}
+              onChange={(e) => {
+                const model: ProductModel = {
+                  ...editProductModel,
+                  lifeCycle: parseInt(e.target.value),
+                };
+                setEditProductModel(model);
+                setReturn(model);
+              }}
+            />
+          </div>
+        </>
+      ),
+      canConfirm:
```

Comment:
feels like using the form library would make this validation logic easier and keep it consistent w/ the order form

Reply:
#395